pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -905,8 +910,23 @@ func (vs *VSphere) CreateVolume(name string, size int, tags *map[string]string)
 	dc, err := f.Datacenter(ctx, vs.cfg.Global.Datacenter)
 	f.SetDatacenter(dc)
 
+	if (*tags)["adapterType"] == "" {
+		(*tags)["adapterType"] = DefaultAdapterType
+	}
+	if (*tags)["diskType"] == "" {
+		(*tags)["diskType"] = DefaultDiskType
+	}
+
+	// vmdks will be created inside kubevols dir
+	kubevolsdir := "[" + vs.cfg.Global.Datastore + "] " + VolDir + "/"
```

Comment:
Hey @dagnello, I looked at #30248 to understand why you wanted to change it. I have fix it, but haven't tested on datastore cluster since I don't have access to one. It does work well with single datastore.